app/browser/reducers/tabsReducer.js

```diff
@@ -55,6 +56,16 @@ const tabsReducer = (state, action, immutableAction) => {
       break
     case appConstants.APP_TAB_UPDATED:
       state = tabState.maybeCreateTab(state, action)
+      const url = action.getIn(['tabValue', 'url'])
+      if (url) {
+        let tabValue = immutableAction.get('tabValue')
+        if (!tabValue) {
+          break
+        }
+        const bookmarked = isLocationBookmarked(state, url)
+        tabValue = tabValue.set('bookmarked', bookmarked)
+        state = tabState.updateTabValue(state, tabValue)
```

Comment:
I think this should all be removed and then just add the bookmarked property in `getTabValue` and it'll work better.